src/openforms/js/components/admin/form_design/logic/ActionSet.js

```diff
@@ -0,0 +1,177 @@
+                // dropdown, but is not used in the backend
+                source: action.componentValueSource
+        }
+    };
+};
+
+const parseJsonAction = (jsonAction) => {
+    let componentValueSource, componentLiteralValue, componentVariableValue;
+    if (jsonAction.action.value.var === undefined) {
+        componentValueSource = 'literal';
+        componentLiteralValue = jsonAction.action.value;
+        componentVariableValue = '';
+    } else {
+        if (!jsonAction.action.value.var.length) {
+            componentValueSource = '';
+        } else {
+            componentValueSource = 'component';
+        }
+        componentLiteralValue = '';
+        componentVariableValue = jsonAction.action.value.var;
+    }
+    if (!!jsonAction.action.source) componentValueSource = jsonAction.action.source;
+
+    return {
+        componentToChange: jsonAction.component,
+        actionType: jsonAction.action.type,
+        componentValueSource: componentValueSource,
+        componentProperty: jsonAction.action.property.value,
+        componentLiteralValue: componentLiteralValue,
+        componentVariableValue: componentVariableValue,
+        componentPropertyValue: jsonAction.action.state,
+    };
+};
+
+const ActionSet = ({name, actions, onChange}) => {
+    const [state, dispatch] = useImmerReducer(reducer, {
+        ...initialState,
+        actions: actions.map(action => parseJsonAction(action)) || []
+    });
+
+    const jsonActions = state.actions.map(action => convertActionToJson(action));
+    const previousJsonActions = usePrevious(jsonActions);
+    useEffect(
+        () => {
+            // if nothing changed, do not fire an update
+            if (previousJsonActions && isEqual(previousJsonActions, jsonActions)) return;
+            onChange({
+                target: {
+                    name: name,
+                    value: jsonActions,
+                }
+            });
+        },
+        [jsonActions]
+    );
+
+    const onActionChange = (index, event) => {
+        const {name, value} = event.target;
+        dispatch({
+            type: 'ACTION_CHANGED',
+            payload: {
+                name,
+                value,
+                index
+            },
+        });
+    };
+
+    const onActionDelete = (index) => {
+        dispatch({
+            type: 'ACTION_DELETED',
+            payload: {index: index}
+        });
+    };
+
+    return (
+        <>
+            {state.actions.map((action, index) => (
+                <Action
+                    key={index}
+                    action={action}
+                    onChange={onActionChange.bind(null, index)}
+                    onDelete={onActionDelete.bind(null, index)}
+                />
+            ))}
+            <button type="button" onClick={ () => dispatch({type: 'ACTION_ADDED'}) }>
+                Add action
```

Comment:
Should be translatable, but probably we need to look at other places where similar buttons/actions are used to have  a consistent UI